pkg/kube/controllers/extendedjob/trigger_reconciler.go

```diff
@@ -105,16 +105,23 @@ func (r *TriggerReconciler) Reconcile(request reconcile.Request) (result reconci
 
 	for _, eJob := range eJobs.Items {
 		if r.query.MatchState(eJob, podState) && r.query.Match(eJob, *pod) {
-			err := r.createJob(ctx, eJob, podName, pod.GetUID())
+			found, err := r.query.IsJobAlreadyExists(ctx, eJob, pod.GetUID(), r)
```

Comment:
I might be missing context here, apparently you found a condition where the trigger reconciler tries to start a job again.
To detect this, we now look at all the pods which have been labeled by the eJob. We check if any pod is the same pod as the one the reconciler has been started with. So since the pod has our label, we skip.

I'm wondering why this triggers at all. The predicates should have stopped this? Do you know more about the conditions which trigger that?

Also while skimming over the code, I'm confused by the label, sometimes it's `ejob-name`, then it's `job-name`. Is that intended, or might that be the bug which triggers additional reconciles?